pkg/log/adapters/leveled.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: This file integrates the logger with retryablehttp.LeveledLogger
+package adapters
+
+import (
+	"context"
+
+	"github.com/getoutreach/gobox/pkg/log"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+// NewLeveledLogger returns a gobox/pkg/log logger that implements the
+// retryablehttp.LeveledLogger interface.
+func NewLeveledLogger(ctx context.Context) retryablehttp.LeveledLogger {
```

Comment:
I'm not a massive fan of this name. I know something like `RetryableHttpLogger` is wordier, but it makes it a little bit more clear what the package is for. Thoughts?

Reply:
yeh Ill go with RetryableHTTPLogger then

Reply:
updated